### DISCUSSION TITLE
Referencing id field value not set when assigning an object

### DISCUSSION THREAD

**Discussion by hoeverma** [serverpod/serverpod#2978] [Q&A] [answered]
Hi!
I have a class defined as follows:

class: InventoryItemType
table: inventory_item_type
fields:
  name: String
  parentId: int?
  parent: InventoryItemType?, relation(field=parentId, optional, onDelete=SetNull)

I thought that, when I have defined objects in my flutter code, like:

InventoryItemType child = InventoryItemType(name: 'typeChild');
InventoryItemType parent = InventoryItemType(name: 'typeParent');

I could do

child.parent = parent;

and the parentId field would be set automatically. Actually, it is not (stays null). Am I doing something wrong, or is this intended? If so, do I always have to set both fields?

Thank you!
Markus

**Comment by SandPod** [comment, marked as answer]
Hey @hoeverma!

Unfortunately, we don't support updating the database reference when assigning the object to the model yet. You will either need to set the field manually or attach the rows in the database and retrieve it from there using the attach row method: https://docs.serverpod.dev/concepts/database/relation-queries#update

Remember to create both objects in the database before you try and create a relationship between them.

The issue for creating nested objects is tracked here: https://github.com/serverpod/serverpod/issues/1343

**Reply by hoeverma** [reply]
Ok, thank you very much for replying so fast!
